### PR TITLE
fix: hide breadcrumbs for deleted pages and show trash

### PR DESCRIPTION
**Pull request by Xazin**
Closes: #5153 

### Feature Preview

https://github.com/user-attachments/assets/08780f7d-e0d2-426d-9f88-c6b2ec961854

#### PR Checklist

- [x] My code adheres to [AppFlowy's Conventions](https://docs.appflowy.io/docs/documentation/software-contributions/conventions)
- [x] I've listed at least one issue that this PR fixes in the description above.
- [ ] I've added a test(s) to validate changes in this PR, or this PR only contains semantic changes.
- [ ] All existing tests are passing.

### CODE REVIEW

File: frontend/appflowy_flutter/lib/workspace/application/view_title/view_title_bar_bloc.dart [hunk 1]
```diff
@@ -30,7 +38,17 @@ class ViewTitleBarBloc extends Bloc<ViewTitleBarEvent, ViewTitleBarState> {
               (s) => s.items,
               (f) => [],
             );
-            emit(state.copyWith(ancestors: ancestors));
+
+            final trash =
+                (await TrashService().readTrash()).toNullable()?.items;
+            final isDeleted = trash?.any((t) => t.id == view.id) ?? false;
```
Comment (LucasXu0) [inline frontend/appflowy_flutter/lib/workspace/application/view_title/view_title_bar_bloc.dart:44]:
```suggestion
            final isDeleted = await trashService.readTrash().fold(
                  (s) => s.items.any((t) => t.id == itemId),
                  (f) => false,
                );
```

**Comment by LucasXu0** [comment]
Including these tests would be better:

- Insert an inline page, delete it, and open it. Check if the trash button is shown in the title bar.
- Recover it from the trash page. Check if the trash button in the title bar is gone.
- Recover it from the banner. Check if the trash button in the title bar is gone.

**Comment by richardshiue** [comment]
I see that this wasn't included in this PR, but I just noticed it. It looks a bit cramped here

Screenshot 2024-10-09 at 11 17 54 AM

**Comment by Xazin** [comment]
> I see that this wasn't included in this PR, but I just noticed it. It looks a bit cramped here
> 
> Screenshot 2024-10-09 at 11 17 54 AM

https://github.com/user-attachments/assets/1fed09a1-a87d-4d92-94bd-ddb26523c3f4

**Comment by richardshiue** [comment]
> > I see that this wasn't included in this PR, but I just noticed it. It looks a bit cramped here
> > Screenshot 2024-10-09 at 11 17 54 AM
> 
> delete.permanentyl.mp4

Got it, nice!

**Pull request merged**